### PR TITLE
[Docs] Add using Git repo as function hub

### PR DESCRIPTION
**Pull request by jillnogold**
[ML-3381](https://jira.iguazeng.com/browse/ML-3381)

### CODE REVIEW

File: docs/runtimes/private-repo.ipynb [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
```
Comment (GiladShapira94) [inline docs/runtimes/private-repo.ipynb:1]: <p>private repo is more relevant for private git repo and how to authenticate to it, can change to - Using your own git repo as a function hub</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (GiladShapira94) [inline docs/runtimes/private-repo.ipynb:1]: <p>The .yaml file will be stored in a path like this: /function-name/function.yaml (e.g /func/function.yaml) - maybe add how it look from the repo point of view, for example: /repo/function-name/function.yaml or https://raw.githubusercontent.com/user name/repo name/function-name/function.yaml"</p><p> </p><p>another point that can be added - in the function folder the one that old the function.yaml file you can stored any additional file for example: source file, notebook example </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: docs/runtimes/git-repo-as-hub.ipynb [hunk 2]
```diff
@@ -0,0 +1,152 @@
+{
```
Comment (GiladShapira94) [inline docs/runtimes/git-repo-as-hub.ipynb:1]: <p>they <s>must</s> be stored in the same folder as the function.yaml - it not mandatory they can be stored in the same folder  </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

**Comment by GiladShapira94** [comment]
Approved 💪

**Review by theSaarco — APPROVED** [review]
This looks good, but I think this is a rather specific use-case and maybe should be documented as an "advanced" topic or something like this. It should be noted that when replacing the hub url then the MLRun marketplace hub will no longer be used, so your repo needs to contain any marketplace functions you may be using.
Also, this will change considerably once we introduce the changes in #3384 to MLRun, which changes how the marketplace functionality works. This is mostly around the implementation of #3384 - we need to make sure this doc is changed accordingly, and also we need to make sure this use-case is still supported with the new mechanism.

**Review by alonmr — APPROVED** [review]

**Pull request merged**